rust/connlib/shared/src/messages.rs

```diff
@@ -131,8 +131,8 @@ impl Eq for RequestConnection {}
 
 #[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq)]
 #[serde(tag = "type", rename_all = "snake_case")]
-pub enum ResourceDescription {
-    Dns(ResourceDescriptionDns),
+pub enum ResourceDescription<TDNS = ResourceDescriptionDns> {
+    Dns(TDNS),
     Cidr(ResourceDescriptionCidr),
 }
```

Comment:
This is somewhat of a hack to not cause an excessive diff. The problem here is that we are using the types from the wire protocol with the portal all the way throughout connlib. I don't think we should do that but instead, have separate types that describe the protocol. Ideally, `firezone-tunnel` doesn't depend on that and instead offers APIs that are called by the upper layers (gateway and client-implementation). Those layers can then integrate with the portal connection.

Alternatively, we could also integrate the portal connection directly with `firezone-tunnel`. That would drastically reduce the API surface which could honestly be a good thing as the entire control protocol would then be an implementation detail.

For now, I chose a path that keeps the diff minimal.

Reply:
Originally, I was planning on having separated types for the tunnel and the control-protocol but I was waiting to implement it with [this issue](https://github.com/firezone/firezone/issues/2187)

> Alternatively, we could also integrate the portal connection directly with firezone-tunnel. That would drastically reduce the API surface which could honestly be a good thing as the entire control protocol would then be an implementation detail.

At this point we'd just get rid of `Tunnel`, right?